homotopy-model/src/proof/signature.rs

```diff
@@ -99,6 +99,29 @@ impl Signature {
             .map_or(0, |id| id + 1)
     }
 
+    #[must_use]
+    pub fn filter_map<F>(&self, f: F) -> Signature
```

Comment:
Could this be moved to homotopy_common? I.e. can this be done for any tree not just signatures?